source/features/new-or-deleted-file.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import * as pageDetect from 'github-url-detection';
+
+import features from '../feature-manager.js';
+import observe from '../helpers/selector-observer.js';
+
+function add(filename: HTMLAnchorElement): void {
+	const list = select('ul[aria-label="File Tree"]');
+	if (!list && pageDetect.isCommit()) {
+		// Silence error, single-file commits don't have the file list
+		return;
+	}
+
+	const fileInList = select(`[href="${filename.hash}"]`, list);
+	if (!fileInList) {
+		features.log.error(import.meta.url, 'Could not find file in sidebar, is the sidebar loaded?');
+		features.unload(import.meta.url);
+		return;
+	}
+
+	const icon = select('.octicon-diff-removed, .octicon-diff-added', fileInList)
+		?.cloneNode(true);
+	if (icon) {
```

Comment:
I added a bunch of conditions to have clear errors in case filelists are deferred in big PRs. I can drop the `features.log.error` condition if anyone finds one such big PR